vowpalwabbit/nn.cc

```diff
@@ -77,12 +77,16 @@ void finish_setup (nn& n, vw& all)
   features& fs = n.output_layer.feature_space[nn_output_namespace];
   for (unsigned int i = 0; i < n.k; ++i)
   { fs.push_back(1., nn_index);
+    std::stringstream ss;
+    ss << "OutputLayer" << i;
```

Comment:
Can you shift lines 80 & 81 into the 'if'?  I'd like auditing to avoid computational impact when not auditing.